src/main/kotlin/com/example/delivery/service/OrderService.kt

```diff
@@ -11,87 +11,92 @@ import com.example.delivery.mapper.OrderMapper.toDomain
 import com.example.delivery.mapper.OrderMapper.toMongoModel
 import com.example.delivery.mapper.OrderMapper.toUpdate
 import com.example.delivery.mapper.OrderWithProductMapper.toDomain
-import com.example.delivery.mapper.ProductMapper.toDomain
 import com.example.delivery.mongo.MongoOrder
 import com.example.delivery.repository.OrderRepository
 import com.example.delivery.repository.ProductRepository
 import com.example.delivery.repository.UserRepository
 import org.bson.types.ObjectId
 import org.springframework.stereotype.Service
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
 
 @Service
 class OrderService(
     private val orderRepository: OrderRepository,
     private val userRepository: UserRepository,
     private val productRepository: ProductRepository,
+    private val productService: ProductService,
 ) {
 
     @LogInvoke
-    fun getById(id: String): DomainOrderWithProduct {
-        return orderRepository.findById(id)?.toDomain()
-            ?: throw NotFoundException("Order with id $id doesn't exists")
+    fun getById(id: String): Mono<DomainOrderWithProduct> {
+        return orderRepository.findById(id)
+            .map { it.toDomain() }
+            .switchIfEmpty(Mono.error(NotFoundException("Order with id $id doesn't exists")))
     }
 
     @SuppressWarnings("ThrowsCount")
-    fun add(createOrderDTO: CreateOrderDTO): DomainOrder {
-        val user = userRepository.findById(createOrderDTO.userId)
-            ?: throw NotFoundException("User with id ${createOrderDTO.userId} doesn't exist")
+    fun add(createOrderDTO: CreateOrderDTO): Mono<DomainOrder> {
+        return userRepository.findById(createOrderDTO.userId)
+            .switchIfEmpty(Mono.error(NotFoundException("User with id ${createOrderDTO.userId} doesn't exist")))
+            .flatMap { user ->
+                val requestedProductsIds = createOrderDTO.items.map { it.productId }.toMutableList()
```

Comment:
you can use reactor-kotlin-extensions 
```suggestion
                val requestedProductsIdsFlux = createOrderDTO.items.map { it.productId }.toFlux()
                // your logic with flux
```